.github/workflows/ci-rust.yml

```diff
@@ -306,3 +318,54 @@
 
             - run: cargo shear
               if: needs.changes.outputs.rust == 'true'
+
+    # Collate job to require a single passing status for branch protection.
+    # After merging, update the repository ruleset (Settings > Rules > Rulesets):
+    #   1. Add "Rust Tests Pass" to required status checks
+    #   2. Remove these 34 individual checks:
+    #      - "Build Rust services"
+    #      - "Lint Rust services"
+    #      - "Shear Rust services"
+    #      - "Test Rust services (*)" (all 31 matrix jobs)
+    #   3. Then apply the TODO comments above to skip jobs entirely when no Rust changes
```

Comment:
We don't need this comment here, make sure you remove it eventually

Reply:
Ah, I like the chaos but it's all good if you wanna/have the time to take it slowly hahah